be/src/util/thrift_rpc_helper.cpp

```diff
@@ -69,27 +89,19 @@ Status ThriftRpcHelper::rpc(const std::string& ip, const int32_t port,
         LOG(WARNING) << "Connect frontend failed, address=" << address << ", status=" << status.message();
         return status;
     }
-    try {
-        try {
-            callback(client);
-        } catch (apache::thrift::transport::TTransportException& e) {
-            status = client.reopen(timeout_ms);
-            if (!status.ok()) {
-                LOG(WARNING) << "client reopen failed. address=" << address << ", status=" << status.message();
-                return status;
-            }
-            callback(client);
-        }
-    } catch (apache::thrift::TException& e) {
-        std::stringstream ss;
-        ss << "call frontend service failed, address=" << address << ", reason=" << e.what();
-        LOG(WARNING) << ss.str();
-        SleepFor(MonoDelta::FromMilliseconds(config::thrift_client_retry_interval_ms * 2));
-        // just reopen to disable this connection
-        (void)client.reopen(timeout_ms);
-        return Status::ThriftRpcError(ss.str());
+
+    status = rpc_impl(callback, client, address);
+    if (status.ok()) {
+        return status;
     }
-    return Status::OK();
+
+    SleepFor(MonoDelta::FromMilliseconds(config::thrift_client_retry_interval_ms));
+    RETURN_IF_ERROR_WITH_WARN(client.reopen(timeout_ms), fmt::format("client reopen failed, address={}:{}", ip, port));
+    status = rpc_impl(callback, client, address);
+    // reopen failure will disable this connection to prevent it from being used again.
+    (void)client.reopen(timeout_ms);
```

Comment:
```suggestion
    if (!status.ok()) {
        // reopen failure will disable this connection to prevent it from being used again.
        (void)client.reopen(timeout_ms);
    }
```

Reply:
need sleep before `reopen`?